transcom/mymove
# MB-3620 Add queries move details page

## Description

> On the MoveDetails page, replace the existing data fetch code with React Query, to fetch and the move order, customer, MTOs, MTO agents, MTO shipments, and MTO service items.
> 
> Notes:
> 
> MoveDetails should be converted from a class component to a functional component
> 
> The queries triggered by componentWillMount should be replaced with useQuery hooks, and Redux connect code should be deleted
> 
> The logic in existing mapStateToProps that maps API data to the format expected by the component may need to be replicated in the useQuery hook.
> 
> Other existing code reliant on the component lifecycle will also need to be modified to use hooks (i.e., the scroll handler and active section state)
> 
> Might be able to use useEffect
> 

> AC:
> 
> - MoveDetails does not rely on Redux for any API data
> 
> - MoveDetails queries the API directly for any data needed based on the move order ID in the URL params

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

**Test**
1. Log in as TOO
2. Click on any new move
3. Click on a shipment and basic service item
4. Click on `Approve selected shipments`
5. Page should refresh
6. Notice approved shipments and service items

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3620) for this change
* [This stackoverflow](https://stackoverflow.com/questions/45006254/jest-mock-module-per-test) shows how we mocked the react-query call and return data for different cases
